pkg/agent/endpoints/metrics.go

```diff
@@ -37,8 +37,8 @@ func (m *connectionMetrics) Preprocess(ctx context.Context, _ string, _ any) (co
 		case middleware.DelegatedIdentityServiceName:
 			adminapi.IncrDelegatedIdentityAPIConnectionCounter(m.metrics)
 			adminapi.SetDelegatedIdentityAPIConnectionGauge(m.metrics, atomic.AddInt32(&m.delegatedIdentityAPIConns, 1))
-		case middleware.HealthServiceName:
-			// Intentionally not emitting metrics for health
+		case middleware.HealthServiceName, middleware.ServerReflectionServiceName, middleware.ServerReflectionV1AlphaServiceName:
```

Comment:
We also need to add the similar exclusion in the `connectionMetrics` `Postprocess()` receiver method.